src/Microsoft.TestPlatform.ObjectModel/Utilities/XmlRunSettingsUtilities.cs

```diff
@@ -209,11 +209,15 @@ public static IXPathNavigable CreateDefaultRunSettings()
             var dataCollectorsNode = doc.CreateElement(Constants.DataCollectorsSettingName);
             dataCollectionRunSettingsNode.AppendChild(dataCollectorsNode);
 
-#if NET451
-            return doc;
-#else
-            return doc.ToXPathNavigable();
+            var ret = doc as IXPathNavigable;
```

Comment:
Since this fix applies only to `!NET451`, could we move the fix accordingly. E.g.
```
#if NET451
    return doc;
#else
   var ret = doc as IXPathNavigable;
   if ...
#endif
```

Please add a comment on why line 212 works for netcoreapp2.0. Can we link to the bug also?